### ISSUE TITLE
Improper filter selections being sent from HL2 Beta 3[BUG]

### ISSUE THREAD

**Issue by N8SDR1** [mi0bot/OpenHPSDR-Thetis#9]
Reid

I will make a video of this and include link to it.

I will also reset my database to exclude and corruption there:

Problem using an HR50 to track band changes with Beta 3- 40-10 meters works correctly and right data is sent.

160,80,60 and (yes, I know not supported 6M) sends wrong band data. 

If I use the standard Thetis version Ritchies the correct band changes and data are sent.

I'll try and get a couple videos of each will need to do video cell phone so I can show HR50 received data.

Wanted to post while had a chance and saw this occurring- will reset database first and send update if that resolves- but you may wish to look at the code/data sent via beta 3. I do not recall this with Beta 2, but it is possible as I was fighting with other issues as well.

Thanks, 
 Rick N8SDR

**Comment by N8SDR1** [comment]
Reid- here are 2 videos showing the error and improper Band data. 

Sorry for the not-so-great quality trying to hold phone, direct mouse and show software selection and the HR50 when switching bands.

https://drive.google.com/drive/folders/1JZdJ7hIHxAcxruJsWsw0VhuwgUrYd2Rj?usp=sharing

**Comment by N8SDR1** [comment]
Reid incase you havent seen this on the Hermes group

G4ZAL
unread,
10:02 AM (2 minutes ago) 
to Hermes-Lite
@ Rick (and maybe FYI Steve too)

Had 2 grandkids here for the day yesterday (child minding over school holidays) and some chores this morning, but checked now and...
HR50 was originally on v3.0D firmware and it had the UNK issue
Updated to v4.1A and it still had the UNK issue, so reflashed back to 3.0D (still gave UNK error) and then HR50 put away as new amp had arrived.

I've put the HR50 back inline and can still reproduce the error...

PowerSDR mRX PS v3.4.9 and Reid's modified Thetis for HL2 will make the HR50 produce a 'row of bricks' on the HR50 display when keying up and then dekeying it shows the UNK error - changing bands, moving the VFO or power cycle usually restores things.
Interestingly, if I reduce the HL2 drive to '0' I still get the error using TUN in PSDR and Thetis
Looking closer at it, when the fault occurs, the 'row of bricks' appear instantly when using TUN button.
Using MOX on SSB doesn't seem to create the fault although VOX will occasionaly fault (full drive set and speaking into the mic).

Quisk seems absolutely fine and no errors at all with HL2 producing 5w into HR50 - using Spot(TUN), PTT(MOX) or VOX - I can't get the UNK error under any circumstance on my test!

This very short test over an hour or so suggests it is software related ?
Put a power meter inline after the HL2 and watch the needle when using the various softwares.
PowerSDR and Thetis both make a short 'blip' of the needle before actual drive seems to be applied (goes along with me noticing the instant error when using TUN)
Quisk does not do this.

Another oddity...
Disconnecting the serial cable and using PSD and Thetis, set the HR50 to use COR for key mode, manually set the band to match the HL2 software band.
I can't seem to get it to error, but watch the HR50 Power/Tx LED - notice it go to Red/Green then Red (and Red stays on) when using TUN - seems the HR50 notices this initial 'blip' of RF.
Connect the serial cable again and PSDR/Thetis will make the UNK error even though I'm using COR !
Once again, Quisk does not do this or make the error no matter how hard I try!

I think it is worth a bit more investigation...

Slightly off track, but same type of cable and connectors used to signal from my HL2 into my Juma PA1000 and at full UK limit of 400w I have no issue, although I have an Arduino Nano reading the HL2 serial output and converting it to BCD for the Juma band data.

Nigel
G4ZAL

**Comment by N8SDR1** [comment]
Also, Reid I wanted to check Harmonics as well- so here is a reply to the Hermes group I posted this morning about Filters and Harmonics.

oday I had a chance to try something a friend suggested-

That was using my other SDR a 2DX with just a piece of coax (about 2 foot long) for the 2DX antenna.

The Hermes was placed in a dummy load.

These radios are on separate power supplies.

I have the drive levels set between 3/4 watt to almost 2 watts in these tests.

The idea was to use the 2DX as a poor man's spectrum display and watch the 2nd Harmonic of the Hermes.

Steve this would not show up on a scope.  Which is why I wanted to look at harmonics from the Hermes- No amp just the Hermes alone.

I have been beating my head since I got the Hermes, I really like it a lot, But I'm plagued with oddities, when running the Hardrock 50 amp where it goes into UNKOWN band errors, and when used to Drive the ACOM I get similar results. 

May initial thinking here was a BIAS issue incorrectly set causing harmonics or spikes (I should have looked at this method of testing first I suppose)

Any way link again to same G Drive folder there are 2 more videos. 

1) Hermes on 40 meters looking at the 2nd Harmonic from the 2DX on 20 Meters
2) Hermes on 80 meters looking at the 2nd Harmonic from the 2DX on 40 Meters

This Does Not Look good; in fact, it could very well explain a lot of the issues I have been having from day 1 again no power amp involved strictly the Hermes PA was used in this test.
If I'm seeing Harmonics like this with a 2nd Receiver no antenna connected to it, I can only Imagine what the Hard Rock 50 and the ACOMs processors are seeing- and thus tripping them with the Errors.

So is it a Bad filter board, Bad Hermes, or something with filter selections in the Thetis software (using Reid's MioBOT Hermes HL2 Beta 3 version)

My next test will be using the Standard Thetis version non HL2 build and perform the same test.

Videos of Harmonics produced from Hermes Lite

https://drive.google.com/file/d/10fF6OUL4nXfd8wxSWg6Uq6sd7tvru5o7/view?usp=sharing   
 (Hermes on 40M looking at 2nd Harmonics produced on 20 Meters) 

https://drive.google.com/file/d/10fakMkKN9ImvXqEhBScTVnf0i9Zx_Bl0/view?usp=sharing  
   (Hermes on 80M looking at 2nd Harmonics produced on 40 Meters)
